clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/SimpleClusterTierClientEntity.java

```diff
@@ -209,6 +222,32 @@ public EhcacheEntityResponse invokeAndWaitForRetired(EhcacheOperationMessage mes
     return invokeInternalAndWait(endpoint.beginInvoke().blockGetOnRetire(true), message, track);
   }
 
+  private void waitForInitCompleted(Duration duration) {
+    boolean interrupted = false;
+    long deadlineTimeout = System.nanoTime() + duration.toNanos();
+    try {
+      while (true) {
+        try {
+          long timeRemaining = deadlineTimeout - System.nanoTime();
+          if (timeRemaining <= 0 || initCompleted) {
+            return;
+          }
+          synchronized (initLock) {
+            if (!initCompleted) {
+              initLock.wait(TimeUnit.NANOSECONDS.toMillis(timeRemaining));
+            }
+          }
+        } catch (InterruptedException e) {
+          interrupted = true;
```

Comment:
For me, interrupting here is like having a smaller timeout. So we should go out of the loop directly. Because it means someone has explicitly called interrupt on this thread. If having a too short timeout breaks everything, than we have another problem.

Reply:
hmm.. if we get out of the loop faster, won't we get into an exception path or a confusing log before the higher layer gets a chance to handle the interrupt, as the init is not done. Also remember that the init itself is a set of local operations, so it is highly unlikely that it will take more than a few micros, unless the machine itself is overloaded. 

Besides this is the same pattern we follow for all invoke messages (see SimpleClusterTierClientEntity)..we make it an uninterruptible wait but then update the interrupt status of the thread. Not sure whether I should change that pattern here (esp, since this is not even a remote operation we are waiting for).

Reply:
Ok... Let's keep it like this and revisit in a more global way later.